plugins/libplugin-pay.c

```diff
@@ -2892,15 +2908,18 @@ static bool routehint_excluded(struct payment *p,
 		 * know the exact capacity we need to send via this
 		 * channel, which is greater than the destination.
 		 */
-		for (size_t j = 0; j < tal_count(hints); j++) {
-			if (!short_channel_id_eq(hints[j].scid.scid, r->short_channel_id))
-				continue;
-			/* We exclude on equality because we set the estimate
-			 * to the smallest failed attempt.  */
-			if (amount_msat_greater_eq(needed_capacity,
-						   hints[j].estimated_capacity))
-				return true;
-		}
+		struct channel_hint *hint;
+		struct short_channel_id_dir scidd;
+		scidd.scid = r->short_channel_id;
+		scidd.dir = node_id_cmp(&routehint->pubkey,
```

Comment:
Since you're iterating through a list of route hints:
```suggestion
		scidd.dir = node_id_cmp(&r->pubkey,
```